src/edu/stanford/nlp/sempre/JsonServer.java

```diff
@@ -0,0 +1,328 @@
+package edu.stanford.nlp.sempre;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import fig.basic.*;
+import fig.html.HtmlElement;
+import fig.html.HtmlUtils;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.file.Paths;
+import java.net.HttpCookie;
+import com.sun.net.httpserver.HttpServer;
+
+import edu.stanford.nlp.sempre.Master.Response;
+import edu.stanford.nlp.sempre.interactive.BadInteractionException;
+
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.Headers;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.ExecutorService;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.time.LocalDateTime;
+
+
+/**
+ * JsonServer, most of the interactive stuff run through this. Handles log instead of master.
```

Comment:
defer: assuming you mean just Server instead of Master. Server has too much html stuff that is not compatible with the interactive work flow, and there is too much code for me to modify effectively.